plugin/trino-google-sheets/src/test/java/io/trino/plugin/google/sheets/TestGoogleSheets.java

```diff
@@ -78,4 +85,48 @@ public void testTableWithRepeatedAndMissingColumnNames()
         assertQuery("desc table_with_duplicate_and_missing_column_names", "SELECT * FROM (VALUES('a','varchar','','')," +
                 " ('column_1','varchar','',''), ('column_2','varchar','',''), ('c','varchar','',''))");
     }
+
+    // test polymorphic table function
+
+    @Test
+    public void testSheetQuerySimple()
+    {
+        assertQuery(
+                "SELECT * FROM TABLE(gsheets.system.sheet(id => '%s'))".formatted(DATA_SHEET_ID),
+                "VALUES " +
+                        "('1', 'one')," +
+                        "('2', 'two')," +
+                        "('3', 'three')," +
+                        "('4', 'four')," +
+                        "('5', 'five')");
+    }
+
+    @Test
+    public void testSheetQueryFilter()
+    {
+        assertQuery(
+                "SELECT * FROM TABLE(gsheets.system.sheet(id => '%s')) WHERE number = '1' and text = 'one'".formatted(DATA_SHEET_ID),
+                "VALUES " +
+                        "('1', 'one')");
+    }
+
+    @Test
+    public void testSheetQueryWithRange()
```

Comment:
Range allows narrowing columns, let's add the test case. e.g. `range => 'number_text!B3:B5'`

Reply:
added.